### PR TITLE
Panic mode arm rotation

### PR DESCRIPTION
**Pull request by MidgleyM**

### CODE REVIEW

File: robotbase/src/main/java/com/team2357/frc2023/commands/PanicModeArmExtender.java [hunk 1]
```diff
@@ -0,0 +1,11 @@
+package com.team2357.frc2023.commands;
```
Comment (NolanCampbell2357) [inline robotbase/src/main/java/com/team2357/frc2023/commands/PanicModeArmExtender.java:1]: delete this

File: robotbase/src/main/java/com/team2357/frc2023/controls/GunnerControls.java [hunk 2]
```diff
@@ -117,7 +118,8 @@ private void mapControls() {
         Trigger yButton = m_yButton.and(noDPad);
         Trigger xButton = m_xButton.and(noDPad);
 
+        upDPadOnly.whileTrue(new ArmRotationCommand(axisRightStickY));
         leftDPadOnly.whileTrue(new ArmExtenderCommand(axisRightStickY));
         aButton.onTrue(new WristToggleCommand());
```
Comment (NolanCampbell2357) [inline robotbase/src/main/java/com/team2357/frc2023/controls/GunnerControls.java:123]: I did not catch this previously. This bind should require the left dpad to be pressed as well.

Comment (NolanCampbell2357) [reply robotbase/src/main/java/com/team2357/frc2023/controls/GunnerControls.java:123]: Add a config for max speed when using controller axis

File: robotbase/src/main/java/com/team2357/frc2023/commands/human/panic/ArmRotationCommand.java [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+    AxisInterface m_axis;
+
+    public ArmRotationCommand(AxisInterface axis) {
+        addRequirements(ArmRotationSubsystem.getInstance());
+        m_axis = axis;
+    }
+
+    @Override
+    public void execute() {
+        ArmRotationSubsystem.getInstance().rotate(m_axis.getValue());
```
Comment (NolanCampbell2357) [inline robotbase/src/main/java/com/team2357/frc2023/commands/human/panic/ArmRotationCommand.java:18]: Subsystem method that you call should have a max speed you multiply the value you pass here by.

**Review by NolanCampbell2357 — CHANGES_REQUESTED** [review]

File: robotbase/src/main/java/com/team2357/frc2023/commands/human/panic/ArmRotationCommand.java [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+    }
+
```
Comment (NolanCampbell2357) [inline robotbase/src/main/java/com/team2357/frc2023/commands/human/panic/ArmRotationCommand.java:20]: Override isFinished and always return false. This is what happens behind the scenes, but we should have it explicitly in the command

File: robotbase/src/main/java/com/team2357/frc2023/commands/human/panic/ArmExtenderCommand.java [hunk 5]
```diff
@@ -2,11 +2,9 @@
```
Comment (NolanCampbell2357) [inline robotbase/src/main/java/com/team2357/frc2023/commands/human/panic/ArmExtenderCommand.java:2]: Override isFinished and always return false. This is what happens behind the scenes, but we should have it explicitly in the command

**Review by NolanCampbell2357 — CHANGES_REQUESTED** [review]

File: robotbase/src/main/java/com/team2357/frc2023/subsystems/ArmExtensionSubsystem.java [hunk 6]
```diff
@@ -3,70 +3,72 @@
...
+    public ArmExtensionSubsystem(CANSparkMax extender, int currentlimit, double ramprate) {
         m_extendMotor = extender;
         m_extendMotor.setSmartCurrentLimit(currentlimit);
         m_extendMotor.setOpenLoopRampRate(ramprate);
         instance = this;
     }
 
     public void configure(Configuration config) {
-        m_config = config;
+        m_config = Constants.ARM_EXTENSION.GET_EXTENSION_CONFIG();
```
Comment (NolanCampbell2357) [inline robotbase/src/main/java/com/team2357/frc2023/subsystems/ArmExtensionSubsystem.java:60]: We should never use constants within subsystems. The config should be set through the subsystem's creation in `SubsystemFactory`

**Review by NolanCampbell2357 — CHANGES_REQUESTED** [review]

**Comment by NolanCampbell2357** [comment]
You need to add a method to create an instance of the arm rotation subsystem to `SubsystemFactory`. That method should then be called in `RobotContainer`. If we ran this on the robot we would get a null pointer exception on the commands due to no instance of the subsystem existing.

File: robotbase/src/main/java/com/team2357/frc2023/subsystems/SubsystemFactory.java [hunk 7]
```diff
@@ -105,4 +107,11 @@ public IntakeArmSubsystem CreateIntakeArmSubsystem() {
                 subsystem.configure(Constants.INTAKE_ARM.GET_INTAKE_ARM_CONFIG());
                 return subsystem;
         }
```
Comment (NolanCampbell2357) [inline robotbase/src/main/java/com/team2357/frc2023/subsystems/SubsystemFactory.java:109]: You need to instantiate the arm extension subsystem as well

Comment (NolanCampbell2357) [reply robotbase/src/main/java/com/team2357/frc2023/subsystems/SubsystemFactory.java:109]: I also do not see a call in `RobotContainer` to `SubsystemFactory` to create the arm rotation subsystem in

**Review by NolanCampbell2357 — DISMISSED** [review]

**Review by Tyson453 — APPROVED** [review]

**Pull request merged**